crates/sui-indexer-alt-framework/src/pipeline/concurrent/collector.rs

```diff
@@ -91,7 +92,11 @@ pub(super) fn collector<H: Handler + 'static>(
         let mut poll = interval(config.collect_interval());
         poll.set_missed_tick_behavior(MissedTickBehavior::Delay);
 
-        // Data for checkpoints that haven't been written yet.
+        // Data for checkpoints that have been received but not yet ready to be sent to committer due to lag constraint.
+        let mut received: BTreeMap<u64, Indexed<H>> = BTreeMap::new();
```

Comment:
I don't think we need to introduce this second "reservoir" -- for sequential pipelines, we store everything in `pending`, and are just selective about what we pop from the pending map. We can do something similar here, and that would simplify things (no need to manage an extra level of hand over between the two buffers, which is tricky to think about when the system is winding down).

I think it suffices to introduce a function like below and call it instead of `pending.first_entry()` on line 121.

```rust
fn next_entry(
    pending: &mut BTreeMap<u64, Pending<H>>, 
    lag: u64,
) -> Option<OccupiedEntry<u64, Pending<H>>> {
    let (&last, _) = pending.last_key_value()?;
    let first = pending.first_entry()?;
    (last - *first.key() >= lag).then(first)
}
```

Reply:
The primary reason I introduced the receiver buffer is to be able to keep the backpressure on pending row count, which I think is still a reasonable back pressure mechanism from the committer, since committer operates based on rows, not checkpoints.

Reply:
We would also have to get rid of MIN_EAGER_ROWS, because we don't actually know how many rows would be ready to be sent to the committer. So these two combined make me realize that change is likely a regression in concept, not an improvement.

Reply:
Okay, sold, that makes sense!